pocs/camera/zwo.py

```diff
@@ -0,0 +1,446 @@
+            else:
+                raise error.PanError("Unepected exposure status on {}: '{}'".format(
+                    self, exposure_status))
+        finally:
+            exposure_event.set()  # write_fits will have already set this, *if* it got called.
+            self._exposure_lock.release()
+
+    def _fits_header(self, seconds, dark):
+        header = super()._fits_header(seconds, dark)
+        header.set('CAM-GAIN', self.gain, 'Internal units')
+        header.set('CAM-BITS', int(get_quantity_value(self._info['bit_depth'], u.bit)),
+                   'ADC bit depth')
+        header.set('XPIXSZ', get_quantity_value(self._info['pixel_size'], u.um), 'Microns')
+        header.set('YPIXSZ', get_quantity_value(self._info['pixel_size'], u.um), 'Microns')
+        header.set('EGAIN', get_quantity_value(self.egain, u.electron / u.adu), 'Electrons/ADU')
+        return header
+
+    def _refresh_info(self):
+        self._info = Camera._ASIDriver.get_camera_property(self._camera_index)
+
+    def _control_getter(self, control_type):
+        if control_type in self._control_info.keys():
+            return Camera._ASIDriver.get_control_value(self._camera_ID, control_type)
+        else:
+            raise NotImplementedError("'{}' has no '{}' parameter".format(self.model, control_type))
+
+    def _control_setter(self, control_type, value):
+        if control_type not in self._control_info.keys():
+            raise NotImplementedError("{} has no '{}' parameter".format(self.model, control_type))
+
+        control_name = self._control_info[control_type]['name']
+        if not self._control_info[control_type]['is_writable']:
+            raise NotImplementError("{} cannot set {} parameter'".format(
+                self.model, control_name))
+
+        if value != 'AUTO':
+            # Check limits.
+            max_value = self._control_info[control_type]['max_value']
+            if value > max_value:
+                msg = "Cannot set {} to {}, clipping to max value {}".format(
+                    control_name, value, max_value)
+                Camera._ASIDriver.set_control_value(self._camera_ID, control_type, max_value)
+                raise error.PanError(msg)
+
+            min_value = self._control_info[control_type]['min_value']
+            if value < min_value:
+                msg = "Cannot set {} to {}, clipping to min value {}".format(
+                    control_name, value, min_value)
+                Camera._ASIDriver.set_control_value(self._camera_ID, control_type, min_value)
+                raise error.PanError(msg)
+        else:
+            if not self._control_info[control_type]['is_auto_supported']:
+                msg = "{} cannot set {} to AUTO".format(self.model, control_name)
+                raise PanError(msg)
+
+        Camera._ASIDriver.set_control_value(self._camera_ID, control_type, value)
+
+    def _setter_try(self, attr, keyword, value):
```

Comment:
The `attr` and `keyword` seem a bit redundant. You use it for `ccd_set_point` and `set_point` but that seems like it's not totally necessary. Is it worth having the extra param just to have that difference in the log output?